### PR TITLE
feat: add sonar-scanner-cli

### PR DESCRIPTION
**Pull request by nickajacks1**
Validated with an enterprise SonarQube instance in CI.

### CODE REVIEW

**Comment by nickajacks1** [comment]
The CI check failed because this version of sonar-scanner-cli was built by a too new version of Java.  `hermit test -t sonar-scanner-cli-6.2.1.4610` executes successfully for me on Ubuntu 24.10 and Ubuntu 24.04.

File: sonar-scanner-cli.hcl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+description = "Scanner CLI for SonarQube"
+repository = "https://github.com/SonarSource/sonar-scanner-cli"
+binaries = ["bin/sonar-scanner"]
+strip = 1
+test = "sonar-scanner --version"
+source = "https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-6.2.1.4610.zip"
```
Comment (alecthomas) [inline sonar-scanner-cli.hcl:6]:
```suggestion
source = "https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${version}.zip"
```

**Review by alecthomas — CHANGES_REQUESTED** [review]

**Comment by alecthomas** [comment]
> The CI check failed because this version of sonar-scanner-cli was built by a too new version of Java. `hermit test -t sonar-scanner-cli-6.2.1.4610` executes successfully for me on Ubuntu 24.10 and Ubuntu 24.04.

Makes sense.

File: sonar-scanner-cli.hcl (hunk 1 above)
Comment (nickajacks1) [reply sonar-scanner-cli.hcl:6]: Oops, thanks.

**Review by alecthomas — APPROVED** [review]

**Pull request merged**